### PR TITLE
[Feature store] Support filtering by date in ingest

### PR DESCRIPTION
**Pull request by katyakats**
https://jira.iguazeng.com/browse/ML-280

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -179,6 +182,9 @@ def ingest(
     :param spark_context: local spark session or True to create one, example for creating the spark context:
                           `spark = SparkSession.builder.appName("Spark function").getOrCreate()`
     :param transformer:   custom transformation function which accepts dataframe and **kwargs as input
+    :param filter_start   datetime, low limit of time needed to be filtered. format '2020-12-01 17:33:15'
+    :param filter_end     datetime, high limit of time needed to be filtered. format '2020-12-01 17:33:15'
```
Comment (Hedingber) [inline mlrun/feature_store/api.py:186]: you're writing datetime but according to the format in the end and your tests I see you're expecting string here
I think we should support both datetime and string here, but if one of them - then datetime
Note that the format you're using is missing the timezone info.
Lastly, I'm not in love with this naming,
what about `start_time` `end_time` and `time_column` ? 
@yaronha WDYT ?

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (yaronha) [reply mlrun/feature_store/api.py:186]: IMO we should be specific and just have start/end time (and the time_key as the column) vs defining generic filter

**Comment by yaronha** [comment]
@katyakats 1. see my comment re doing start/end time filter and not generic, 2. for this PR to be useful we need to add support for start/end time in the `local merger`  (get offline features flow), see: https://github.com/mlrun/mlrun/blob/0e109005281d714542012055e913fd55c3e51266/mlrun/feature_store/retrieval/local_merger.py#L31

the `df = feature_set.to_dataframe(columns=column_names, df_module=df_module)` need to read the data in the specified time window

**Comment by katyakats** [comment]
@yaronha i will add filtering in offline features flow in a separate commit

**Pull request merged**